volplugin/volplugin.go

```diff
@@ -95,3 +110,45 @@ func logHandler(name string, debug bool, actionFunc func(http.ResponseWriter, *h
 		actionFunc(w, r)
 	}
 }
+
+func (dc *DaemonConfig) updateMounts() error {
+	cd := ceph.NewDriver()
+	mounts, err := cd.Mounted()
+	if err != nil {
+		return err
+	}
+
+	for _, mount := range mounts {
+		parts := strings.Split(mount.Volume.Name, "/")
+		if len(parts) != 2 {
+			log.Warnf("Invalid volume named %q in ceph scan: skipping refresh", mount.Volume.Name)
+			continue
+		}
+
+		log.Infof("Refreshing existing mount for %q", mount.Volume.Name)
+
+		volConfig, err := requestVolumeConfig(dc.Master, parts[0], parts[1])
+		if err != nil {
```

Comment:
Hmm. I actually totally overlooked the volmaster down state. I will look into dividing these errors up, it will greatly clarify what needs to be done as you have suggested.

Another condition is when the volume didn't exist in the first place, but is registered as a legal mount point we detected at startup. This is much harder to deal with as there's no accessible journal of when it changed. I'm not entirely sure if we should be handling those at all or taking action to reconcile the system with what we perceive is reality by removing the mounts.